pandas/tseries/index.py

```diff
@@ -1629,13 +1632,17 @@ def normalize(self):
         return DatetimeIndex(new_values, freq='infer', name=self.name,
                              tz=self.tz)
 
-    def searchsorted(self, key, side='left'):
-        if isinstance(key, (np.ndarray, Index)):
-            key = np.array(key, dtype=_NS_DTYPE, copy=False)
+    @Substitution(klass='DatetimeIndex')
+    @Appender(_shared_docs['searchsorted'])
+    def searchsorted(self, v, side='left', *args):
+        validate_args(args, min_length=0, max_length=1)
```

Comment:
@jreback : Regarding args that are accepted in the `numpy` implementation but are extraneous to the `pandas` implementation`, where do you draw the boundary between throwing when they are non-null and not (i.e. we throw when `out != None` but not when `sorter != None` as was proposed by @shoyer)?